modules/translator.py

```diff
@@ -89,7 +98,31 @@ def get_gpt_translation(self, user_prompt: str, model: str, system_prompt: str,
 
         translated = response.choices[0].message.content
         return translated
-    
+
+    def get_local_gpt_translation(self, user_prompt: str, model: str, system_prompt: str, image: np.ndarray):
+        encoded_image = encode_image_array(image)
+
+        if self.img_as_llm_input:
+            message = [
+                    {"role": "system", "content": [{"type": "text", "text": system_prompt}]},
+                    {"role": "user", "content": [{"type": "text", "text": user_prompt}, {"type": "image_url", "image_url": {"url": f"data:image/png;base64,{encoded_image}"}}]}
+                ]
+        else:
+            message = [
+                    {"role": "system", "content": system_prompt},
+                    {"role": "user", "content": user_prompt}
+                ]
```

Comment:
This may be a Jan-specific issue, but requests in format: `{"type": "text", "text": system_prompt}` do not work, since only text is supported (images are not). However, I left the image option toggle and the request format for that alone, in case any OpenAI compatible servers _do_ support images.